### ISSUE TITLE
Bunker Unload Not Following path.

### ISSUE THREAD

**Issue by Gibbanator** [Courseplay/courseplay#6749]
I am not sure what version this started happening but When I made a new course to unload my bunker to set up a path to unload both bunkers at same time the Tractor will not follow the path set. When I bring up the Error panel it show some strange stuff. Here is a video on the problem that also shows errors.
https://youtu.be/iBjsXfgT8s8

added log txt
[log.txt](https://github.com/Courseplay/courseplay/files/5890091/log.txt)

**Comment by Gibbanator** [comment]
update.. After I sent log text to you I decided to look at it and I noticed that there were errors in there for some mods... I decided to remove those mods from my mod folders.. Downloaded and installed the new CP version 00033, Logged back into my save and made a new course. It is working fine now other that it overshooting the waypoint at the unload point it is still working good.

I am not sure which step I did to fix the problem but I did not delete this hoping that others may see a fix if they have same problem.

I would like to see the overshooting the unload waypoint fixed in future updates. 

CP is best for large farming and could not play without it..

**Comment by Tensuko** [comment]
What you see in the log should be another driver. It is only the W190D the other one is an Unloader.

Your course is wrong too. Stop must be in front of the start point, and if you maybe have the Stop point placed in reverse, then I know why this happens.
![mode9](https://user-images.githubusercontent.com/27344744/106245852-7b708280-620d-11eb-9010-2e383a0b3607.png)

**Comment by Tensuko** [comment]
Just letting my Mode9 driver run with Version .00033 and no Issue at all.
![fsScreen_2021_01_29_11_35_00](https://user-images.githubusercontent.com/27344744/106264573-1e34fb00-6226-11eb-9da0-fc4401eab29f.png)

**Comment by Gibbanator** [comment]
my stop point was set after I put the course back to forward and it was behind the start point as shows in Video; the only thing I did not do was move forward after making the stop point (but put back forward before; making stop point). Also Waypoint at Unload point I had put while in forward position (Should I put it in reverse before making the waypoint?). Could that be the reason I am overshooting the waypoint at the Unload point?

When I look at your Bunker course from your screenshot I only see see one waypoint (Set in front on bunker.) 

So my setting are:

1,Start course and move forward
2.set Waypoint front of Bunker
3.move forward to bank of bunker set waypoint **_(while in forward position)_**
4.Put in reverse, back out of bunker (with turn to right of bunker)
5.Put in forward position and move to Unload point set waypoint **_(while in forward position)_**
6.Put in reverse move to area behind start point and put in forward position and hit STOP.

Am I in the right position when I set the waypoints?
Should I move forward a bit after setting the stop point? (There is not much room to do so)

**Comment by Tensuko** [comment]
> 6.Put in reverse move to area behind start point and put in forward position and hit STOP.

You forgot to drive forward towards the start point and THEN hit stop.
Your Stoppoint is out of position.

**Comment by Gibbanator** [comment]
okay I made new courses but now doing as in Video. I have no ideal what I am doing wrong. I am doing exactly as I described to you above including moving forward to start point. Here I made another video: 
https://youtu.be/07vTDHNdncM

In Video I make the course and run it but it does not follow paths.. Did I make the course wrong?
[log.txt](https://github.com/Courseplay/courseplay/files/5894149/log.txt)

**Comment by Tensuko** [comment]
https://youtu.be/er2QKoOKkxk

**Comment by Gibbanator** [comment]
Thanks Tensuko.. I am somewhat confused. You make your course the same way I did with the all the waypoints and stop. You did edit your end bunker waypoint and moved it to the end of bunker. I made my new course as you showed in your video and I edit the the bunker back way point as you did but I also had to move the reverse waypoint to the same point for it to work.. but it is working but I do not understand why yours worked and mine did not with my prior course. maybe it was the edit.

Even with it working it is still overshooting the unload waypoint.. Please not in video:
https://youtu.be/7FSWA-Bgqyw
lol I am so confused why yours work perfectly and I can not get mine to work as yours.

**Comment by Tensuko** [comment]
Do you know how the CourseManager.xml works ?
\My Games\FarmingSimulator2019\CoursePlay_Courses\MapUS -> courseManager.xml
Add in this line (maybe give it another Number, if you don't want to overwrite existing one)
<slot fileName="courseStorage0002.xml" isUsed="true" id="28" parent="0" name="Mode9_TEST BGA"/>
And paste the .xml (unzip it) in your folder as well. (same name as in the line).
[courseStorage0002.zip](https://github.com/Courseplay/courseplay/files/5895017/courseStorage0002.zip)

Try my course. I would be even more confused if that one is working for you then.
There still can be a modconflict.

**Comment by Gibbanator** [comment]
I added your course and edited the courseManager.xml to match.

It works perfectly.. So no mod conflict.... I checked your course in edit mode and your Waypoints and stop area is same as my course buy my course overshoots unloading waypoint.. I do not know why it does that. I will use your course of course but I it will occur again when I make new courses I am sure.. Which I knew what I am doing wrong...
Here is Mine.. I see no errors in the log.txt
[courseStorage0008.zip](https://github.com/Courseplay/courseplay/files/5895304/courseStorage0008.zip)

**Comment by Tensuko** [comment]
Will have a look later.

**Comment by envikowaty** [comment]
Tensuko i just have question, is possible to use this course (Mode9) for loading Wilson trailer with two separate boxes ?

**Comment by Tensuko** [comment]
Yes, that should be possible, but we are also working on some adjustments for this.

**Comment by envikowaty** [comment]
> Yes, that should be possible, but we are also working on some adjustments for this.

Based on your movie all working fine with all loders in game. I'm looking forward for the option for Wilson trailer 👍

**Comment by envikowaty** [comment]
small but: all was fine till the moment when I restarted the game

![Clipboard01](https://user-images.githubusercontent.com/69640077/106361372-e315f300-631d-11eb-9952-5e3485e3c9b1.jpg)

https://gist.github.com/envikowaty/f4f4255e45e390aad5fff1c708c04ed0

**Comment by envikowaty** [comment]
@Tensuko the problem which I posted higher is because of mod ICONIK XL bucket, mod delete all working fine

**Comment by Tensuko** [comment]
What about the origin of this Issue ? Can we take it as solved for now ?
Unless we exactly know whats going on, we can't do anything.

**Comment by envikowaty** [comment]
At The moment is ok, still collecting corn for next tests

**Comment by Gibbanator** [comment]
at the unload point on my course there is a back and forward point together. If I move the forward point back the vehicle will stop at the waypoint at the unload point. I find that I have to edit the points to the unload waypoint to alternate forward and backward points that are inline to the Unload point.. This was the only way I could get it to work.

All the bunker courses I make always have the bank and forward points at the unload point (together) and is why it will overshoot that point. Will only work with the edit. So it looks like for me at least I have to edit my course to get them to work properly.

If this is unclear I can make another video to show the edit.

**Comment by Crinouche** [comment]
There are issues in this topic similar to
# 6447. I even think # 6734 is related

**Comment by Crinouche** [comment]
There are issues in this topic similar to # 6447. I even think # 6734 is related

**Comment by Gibbanator** [comment]
I made several new courses to test and putting it in reverse at unloading point does not work. Vehicle does not follow path or will over shoot Unloading waypoint. I redid the courses with forward setting than placing waypoint at unload point But if I edit the course as I described they all work.

Here is video on my edit.. Moving the forward waypoint back I can get my course to work. https://youtu.be/BOPkvAjFy8o
